recipes/mpg123/all/conanfile.py

```diff
@@ -60,29 +62,37 @@ def configure(self):
         del self.settings.compiler.cppstd
         if self.options.shared:
             del self.options.fPIC
+
+    def requirements(self):
+        if self.options.module == "libalsa":
+            self.requires("libalsa/1.2.4")
+        if self.options.module == "tinyalsa":
+            self.requires("tinyalsa/1.1.1")
+
+    def validate(self):
         try:
             int(self.options.seektable)
         except ValueError:
             raise ConanInvalidConfiguration("seektable must be an integer")
         if self.settings.os != "Windows":
-            if self.options == "win32":
+            if self.options.module == "win32":
                 raise ConanInvalidConfiguration("win32 is an invalid module for non-Windows os'es")
 
-    def requirements(self):
-        if self.options.module == "libalsa":
-            self.requires("libalsa/1.2.4")
-        if self.options.module == "tinyalsa":
-            self.requires("tinyalsa/1.1.1")
+    @property
+    def _settings_build(self):
+        return self.settings_build if hasattr(self, "settings_build") else self.settings
```

Comment:
What do you think about this syntax?
```suggestion
        return getattr(self, "settings_build", self.settings)
```